ObjectiveGit/GTRepository.m

```diff
@@ -919,4 +920,80 @@ - (BOOL)cleanupStateWithError:(NSError * _Nullable __autoreleasing *)error {
 	return YES;
 }
 
+#pragma mark Notes
+
+- (GTNote *)createNote:(NSString *)note target:(GTObject *)theTarget referenceName:(NSString *)referenceName author:(GTSignature *)author committer:(GTSignature *)committer overwriteIfExists:(BOOL)overwrite error:(NSError **)error {
+	git_oid oid;
+	
+	int gitError = git_note_create(&oid, self.git_repository, referenceName.UTF8String, author.git_signature, committer.git_signature, theTarget.OID.git_oid, [note UTF8String], overwrite ? 1 : 0);
+	if (gitError != GIT_OK) {
+		if (error != NULL) *error = [NSError git_errorFor:gitError description:@"Failed to create a note in repository"];
+		
+		return nil;
+	}
+	
+	return [[GTNote alloc] initWithTargetOID:theTarget.OID repository:self referenceName:referenceName error:error];
+}
+
+- (BOOL)removeNoteFromObject:(GTObject *)parentObject referenceName:(NSString *)referenceName author:(GTSignature *)author committer:(GTSignature *)committer error:(NSError **)error {
+	int gitError = git_note_remove(self.git_repository, referenceName.UTF8String, author.git_signature, committer.git_signature, parentObject.OID.git_oid);
+	if (gitError != GIT_OK) {
+		if(error != NULL) *error = [NSError git_errorFor:gitError description:@"Failed to delete note from %@", parentObject];
+		return NO;
+	}
+	
+	return YES;
+}
+
+- (BOOL)enumerateNotesWithReferenceName:(NSString *)referenceName error:(NSError **)error usingBlock:(void (^)(GTNote *note, GTObject *object, BOOL *stop))block
+{
+	git_note_iterator *iter = NULL;
+	
+	int gitError = git_note_iterator_new(&iter, self.git_repository, referenceName.UTF8String);
+	
+	if (gitError != GIT_OK)
+	{
+		if(error != NULL) *error = [NSError git_errorFor:gitError description:@"Failed to enumerate notes"];
+		return NO;
+	}
+	
+	git_oid note_id;
+	git_oid object_id;
+	BOOL success = YES;
+	int iterError = GIT_OK;
+	
+	while ((iterError = git_note_next(&note_id, &object_id, iter)) == GIT_OK) {
+		NSError* lookupErr = nil;
+		
+		GTNote* note = [[GTNote alloc] initWithTargetGitOID:&object_id repository:self.git_repository referenceName:referenceName.UTF8String error:&gitError];
+		if (note == nil) {
+			if (error != NULL) *error = [NSError git_errorFor:gitError description:@"Cannot find a note"];
+			success = NO;
+			break;
+		}
+		
+		GTObject* obj = [self lookUpObjectByGitOid:&object_id error:&lookupErr];
```

Comment:
I'll see how this can be reworked to be in sync with #590. But I'll submit a different pull request for that one.